ui/pages/confirmation/util.ts

```diff
@@ -0,0 +1,155 @@
+import { ResultComponent } from '@metamask/approval-controller';
+
+type TemplateRendererComponent = {
+  key: string;
+  element: string;
+  props?: Record<string, unknown>;
+  children?:
+    | string
+    | TemplateRendererComponent
+    | (string | TemplateRendererComponent)[];
+};
+
+/**
+ * Processes an error message or ResultComponent and returns a TemplateRendererComponent
+ * or an array of strings | TemplateRendererComponents.
+ *
+ * @param input - The message or component to process.
+ * @param fallback - The fallback message to use when the input is not valid.
+ * @returns The processed error component.
+ */
+export function processError(
+  input: undefined | string | ResultComponent | ResultComponent[],
+  fallback: string,
+): TemplateRendererComponent | (string | TemplateRendererComponent)[] {
+  let currentInput = convertResultComponents(input);
+
+  if (!currentInput) {
+    currentInput = fallback;
+  }
+
+  if (!(typeof currentInput === 'string')) {
+    return currentInput;
+  }
+
+  return {
+    key: 'error',
+    element: 'ActionableMessage',
+    props: { type: 'danger', message: currentInput },
+  };
+}
+
+/**
+ * Processes a string or ResultComponent and returns a string or TemplateRendererComponent
+ * or an array of strings | TemplateRendererComponents.
+ *
+ * @param input - The message or component to process.
+ * @param fallback - The fallback string to use when the input is not valid.
+ * @returns The processed message.
+ */
+export function processString(
+  input: undefined | string | ResultComponent | ResultComponent[],
+  fallback: string,
+): string | TemplateRendererComponent | (string | TemplateRendererComponent)[] {
+  let currentInput = convertResultComponents(input);
+
+  if (!currentInput) {
+    currentInput = fallback;
+  }
+
+  if (!(typeof currentInput === 'string')) {
+    return currentInput;
+  }
```

Comment:
this is exactly the same as in the `processError`.. can't we abstract it in its own method?

Reply:
I've applied your suggestions. Not including the abstraction once I tried I didn't see much gain on it.